src/software/world/timestamped_robot_state_test.cpp

```diff
@@ -78,3 +78,37 @@ TEST(RobotStateWithTimestampTest, compare_states_with_different_timestamps)
     EXPECT_TRUE(state_1 == state_2);
     EXPECT_FALSE(state_1 != state_2);
 }
+
+TEST(RobotStateWithTimestampTest, update_with_set_robot_state)
+{
+    TimestampedRobotState timestamped_robot_state_1(
+        Point(-1.2, 3), Vector(2.2, -0.05), Angle::quarter(),
+        AngularVelocity::fromRadians(1.1), Timestamp::fromSeconds(123));
+    TimestampedRobotState timestamped_robot_state_2(
+        Point(-1.2, 3), Vector(2.2, -0.05), Angle::quarter(),
+        AngularVelocity::fromRadians(1.1), Timestamp::fromSeconds(123));
+    EXPECT_EQ(timestamped_robot_state_1, timestamped_robot_state_2);
+
+    timestamped_robot_state_2.robotState().setBallInMouth(true);
+    timestamped_robot_state_2.robotState().setTimeSinceLastChip(
+        Duration::fromMilliseconds(11));
+    timestamped_robot_state_2.robotState().setTimeSinceLastKick(
+        Duration::fromMilliseconds(5));
+    EXPECT_FALSE(timestamped_robot_state_2.robotState().timeSinceLastChip());
+    EXPECT_FALSE(timestamped_robot_state_2.robotState().timeSinceLastKick());
+    EXPECT_EQ(timestamped_robot_state_1, timestamped_robot_state_2);
+
+    RobotState updated_state = timestamped_robot_state_2.robotState();
+    updated_state.setBallInMouth(true);
+    updated_state.setTimeSinceLastChip(Duration::fromMilliseconds(11));
+    updated_state.setTimeSinceLastKick(Duration::fromMilliseconds(5));
+    timestamped_robot_state_2.setRobotState(updated_state);
+
+    EXPECT_TRUE(timestamped_robot_state_2.robotState().timeSinceLastChip());
+    EXPECT_TRUE(timestamped_robot_state_2.robotState().timeSinceLastKick());
```

Comment:
These should be `ASSERT_TRUE`, otherwise the test will segfault later.

Reply:
I don't actually need to derefence, I can compare std::optional to the value directly